tests/tests/mod.rs

```
@@ -0,0 +1,53 @@
+// Copyright 2018 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+mod utils;
+
+use self::utils::{init_rng, FullId, Network, Transaction};
+use parsec::SecretId;
+use std::collections::BTreeSet;
+
+#[test]
+fn minimal_network() {
+    let mut rng = init_rng(None);
+
+    // 4 is the minimal network size for which the super majority is less than it.
+    let num_peers = 4;
+    let max_iterations = 100;
+
+    let mut network = Network::new(num_peers);
+    let id = FullId::new();
+    let transaction = Transaction::InsertPeer(*id.public_id());
+
+    for peer in network.peers.values_mut() {
+        assert!(!peer.have_voted_for(&transaction));
+        assert!(peer.vote_for(transaction.clone()).is_ok());
+    }
+
+    let mut consensued_peer = BTreeSet::new();
+    let mut iterations = 0;
+    loop {
+        network.send_random_syncs(&mut rng);
+        for (peer_id, peer) in &mut network.peers {
+            if let Ok(result) = peer.poll() {
```

I think we should unwrap the result here rather than silently swallowing the error.